### PR TITLE
mysql-shell: resolve locale-setting error

### PR DESCRIPTION
**Pull request by aaronjheng**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/mysql-shell/default.nix [hunk 1]
```diff
@@ -114,7 +115,8 @@ stdenv.mkDerivation rec{
   CXXFLAGS = [ "-DV8_COMPRESS_POINTERS=1" "-DV8_31BIT_SMIS_ON_64BIT_ARCH=1" ];
 
   postFixup = ''
-    wrapProgram $out/bin/mysqlsh --set PYTHONPATH "${pythonPath}"
+    wrapProgram $out/bin/mysqlsh --set PYTHONPATH "${pythonPath}" \
```
Comment (SuperSandro2000) [inline pkgs/development/tools/mysql-shell/default.nix:118]:
```suggestion
    wrapProgram $out/bin/mysqlsh \
      --set PYTHONPATH "${pythonPath}" \
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
But why is this required? This variable is set on NixOS and on other systems needs to be managed by something like home-manager.

**Comment by aaronjheng** [comment]
I'm using nix on Ubuntu. Runing mysql-shell without `LOCALE_ARCHIVE` set, I get the following warning:
```
Cannot set LC_ALL to locale en_US.UTF-8: No such file or directory
```

**Comment by SuperSandro2000** [comment]
Then this must be fixed in your installation because this problem will occur with way more packages.

**Comment by aaronjheng** [comment]
The following packages use the same approach:
* mixxx
* sublimetext3-bin
* sublimetext4-bin
* cplex
* sublime-merge
* guake
* fluxboxlauncher

@SuperSandro2000

**Pull request closed without merging**